trinity/chains/__init__.py

```diff
@@ -56,6 +65,50 @@
 )
 
 
+def validate_genesis(chain_config: ChainConfig) -> (BlockHeader, int):
+    """
+    Checks the genesis configuration file provided and ensures that it is valid.
+    """
+    if not os.path.exists(chain_config.genesis):
+        raise MissingPath(
+            "The base chain genesis configuration file does not exist: `{0}`".format(
+                chain_config.genesis,
+            ),
+        )
+
+    with open(chain_config.genesis, 'r') as genesis_config:
+        raw_genesis = genesis_config.read()
+        genesis = json.load(raw_genesis)
+
+        if not 'chainId' in genesis.keys():
+            raise ValidationError("genesis config missing required 'chainId'")
+        if not 'difficulty' in genesis.keys():
+            raise ValidationError("genesis config missing required 'difficulty'")
+        if not 'gasLimit' in genesis.keys():
+            raise ValidationError("genesis config missing required 'gasLimit'")
+        if not 'nonce' in genesis.keys():
+            raise ValidationError("genesis config missing required 'nonce'")
+        if not 'extraData' in genesis.keys():
+            raise ValidationError("genesis config missing required 'extraData'")
+
+        return BlockHeader(
```

Comment:
Our general pattern is that `validate_` functions don't return data, so this should likely be re-written to three independent things.

1. load the JSON
2. validate the JSON
3. extraction (block header, chain configuration, etc)